core/src/main/scala/kafka/utils/CoreUtils.scala

```diff
@@ -77,7 +77,17 @@ object CoreUtils extends Logging {
    * @param log The log method to use for logging. E.g. logger.warn
    * @param action The action to execute
    */
-  def swallow(log: (Object, Throwable) => Unit, action: => Unit) {
+  def swallow(log: (String, Throwable) => Unit, action: => Unit) {
+    swallow(action)(log)
+  }
+
+  /**
+    * Do the given action and log any exceptions thrown without rethrowing them.
+    * @param action The action to execute
+    * @param log The log method to use for logging. E.g. logger.underlying.warn. This
+    *            parameter can be used as an implicit parameter.
+    */
+  def swallow(action: => Unit)(implicit log: (String, Throwable) => Unit) {
```

Comment:
I'll attach some explanation for the implicits.
So when I was removing the `Logging.swallow` methods, I realized that it has quite an extensive usage through `CoreUtils`. In some places, like `KafkaServer.shutdown` it is used heavily and therefore using `CoreUtils.swallow` directly would decrease readability as you'd have to specify the logger function with every call. Therefore I refactored the swallow function a bit, to use implicit parameter and I declare the implicit field in the class itself. This way I can maintain readability but complicate the code a bit. What do you think about this @ijuma? Is implicits something that we try to avoid in Kafka or is it fine to use them?

Reply:
Generally, it's best to avoid implicits for types like this (they're not specific to a domain). My suggestion is to have a signature like:

```
def swallow(action: => Unit, logger: Logging, logLevel: Level = Level.WARN)
```

We could make the logger implicit, but it will generally be just `this` so it won't help.

Reply:
Ok, I'll refactor it.